h2/src/main/org/h2/mvstore/db/MVTable.java

```diff
@@ -754,13 +759,12 @@ public void lockRows(Session session, Iterable<Row> rowsForUpdate) {
 
     private void analyzeIfRequired(Session session) {
         synchronized (this) {
-            if (nextAnalyze == 0 || nextAnalyze > changesSinceAnalyze++) {
+            if (++changesSinceAnalyze <= nextAnalyze) {
```

Comment:
Why logic was changed here? Initial value of `nextAnalyze` is copied from `ANALYZE_AUTO` that has a special meaning for 0 that means to not automatically run `ANALYZE`.

Reply:
It goes along with reassignment  of the special value [here](https://github.com/h2database/h2database/blob/a4ed2d4c1b6dcc817df3697aa118f9f3fe14eb57/h2/src/main/org/h2/mvstore/db/MVTable.java#L132), so you do not have to check every time.

Reply:
On a higher traffic table, this is eventually going to start analyzing again. If you're aiming for performance, changing to an AtomicInteger and skipping the synchronized would make more difference.

Reply:
You are absolutely right, will do just that